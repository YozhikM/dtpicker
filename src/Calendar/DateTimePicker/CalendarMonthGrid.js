/* @flow */

import React from 'react';
import {
  startOfMonth,
  endOfMonth,
  eachDay,
  getDay,
  isSameDay,
  isWithinRange,
  getDate,
} from 'date-fns';
import s from './Calendar.scss';
import CalendarDay from './CalendarDay';
import format from './format';
import { convertToUTCNewDate, getDateFromValue, type Highlight } from '../helpers';

type Value = {
  year: number,
  month: number,
};

type Week = number[];

export type Props = {|
  value: Value,
  highlight?: Highlight[] | Highlight,
  maxDate?: Date,
  minDate?: Date,
  onSetDate?: (day: number) => void,
  onClickMonth?: (month: Value) => void,
|};

type State = {
  weeks: Week[],
};

export default class CalendarMonthGrid extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      weeks: this.createMonth(props.value) || [],
    };
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.value !== this.props.value) {
      this.setState({ weeks: this.createMonth(nextProps.value) });
    }
  }

  createMonth(date: Value): Week[] {
    const MONTH = new Date(date.year, date.month);
    const startDateOfMonth = startOfMonth(MONTH);
    const endDateOfMonth = endOfMonth(MONTH);
    const currentMonth: number[] = eachDay(startDateOfMonth, endDateOfMonth).map(day =>
      getDate(convertToUTCNewDate(day))
    );

    const firstOfMonthDay: number = getDay(startDateOfMonth);
    let beginOfMonth;
    if (firstOfMonthDay === 0) {
      beginOfMonth = 1;
    } else {
      beginOfMonth = 8 - firstOfMonthDay;
    }
    const firstWeek = currentMonth.slice(0, beginOfMonth);
    const secondWeek = currentMonth.slice(beginOfMonth, 7 + firstWeek.length);
    const thirdWeek = currentMonth.slice(beginOfMonth + 7, beginOfMonth + 14);
    const fourthWeek = currentMonth.slice(beginOfMonth + 14, beginOfMonth + 21);
    const fiveWeek = currentMonth.slice(beginOfMonth + 21, beginOfMonth + 28);
    const tailWeek = currentMonth.slice(beginOfMonth + 28);
    const weeks: Week[] = [firstWeek, secondWeek, thirdWeek, fourthWeek];
    // check for february starting on monday, out of 28 days
    if (fiveWeek.length > 0) {
      weeks.push(fiveWeek);
    }
    if (tailWeek.length > 0) {
      weeks.push(tailWeek);
    }
    return weeks;
  }

  onSetDate = (day: number) => {
    const { maxDate, minDate, onSetDate, value } = this.props;
    const date = new Date(value.year, value.month, day);

    if ((maxDate && date > maxDate) || (minDate && date < minDate)) return;

    if (onSetDate) {
      onSetDate(day);
    }
  };

  onClickMonth = () => {
    const { onClickMonth, value } = this.props;
    if (onClickMonth) {
      onClickMonth(value);
    }
  };

  getWeekDays(): string[] {
    const { value } = this.props;
    const { weeks } = this.state;

    return weeks[1].map(weekDay => {
      const date = new Date(value.year, value.month, weekDay);
      return format(date, 'dd');
    });
  }

  setDayStyle = (highlight: Highlight[] | Highlight, day: number) => {
    const { maxDate, minDate, value } = this.props;
    const date = new Date(value.year, value.month, day);

    const highlightStyle = { backgroundColor: '#34495e', color: '#fff' };
    const rangeStyle = { backgroundColor: '#8196ab', color: '#fff' };
    const disabledDateStyle = { color: 'lightgrey' };

    if (Array.isArray(highlight)) {
      const [first, second] = highlight;
      const firstDate = getDateFromValue(first);
      const secondDate = getDateFromValue(second);

      if ((maxDate && date > maxDate) || (minDate && date < minDate)) {
        return disabledDateStyle;
      }

      for (let i = 0; i < highlight.length; i++) {
        if (isSameDay(getDateFromValue(highlight[i]), date)) {
          return highlightStyle;
        }
      }
      if (firstDate < secondDate) {
        if (isWithinRange(date, firstDate, secondDate)) {
          return rangeStyle;
        }
      } else if (isWithinRange(date, secondDate, firstDate)) {
        return rangeStyle;
      }
    } else if (isSameDay(getDateFromValue(highlight), date)) {
      return highlightStyle;
    }

    return {};
  };

  render() {
    const { weeks } = this.state;
    const { value, highlight } = this.props;
    const weekDays = this.getWeekDays();
    const month = new Date(value.year, value.month);

    return (
      <div className={s.root}>
        <span className={s.span} onClick={this.onClickMonth}>
          {format(month, 'MMMM YYYY')}
        </span>
        <table>
          <thead>
            <tr>
              {weekDays.map(weekDay => {
                return <th key={weekDay}>{weekDay.toUpperCase()}</th>;
              })}
            </tr>
          </thead>

          <tbody>
            {weeks.map(week => (
              <tr key={week[0]}>
                {week.map(day => {
                  return (
                    <CalendarDay
                      key={day}
                      day={day}
                      style={highlight ? this.setDayStyle(highlight, day) : {}}
                      onClick={this.onSetDate}
                    />
                  );
                })}
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}
