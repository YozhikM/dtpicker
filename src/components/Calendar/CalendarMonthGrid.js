/* @flow */

import React from 'react';
import s from './Calendar.scss';
import CalendarDay from './CalendarDay';
import { startOfMonth, endOfMonth, eachDay, getDay, isSameDay, isWithinRange } from 'date-fns';
import format from './format';

export type Props = {|
  value: Date,
  // false - disable highlight
  // null or undefined - use date for highlight
  // value - use provided value(s) for highlight
  highlight?: Array<Date> | Date,
  onSetDate?: (date: Date) => void,
  onClickMonth?: (date: Date) => void
|};

type State = {|
  weeks: Array<Array<Date>>
|};

class CalendarMonthGrid extends React.Component<Props, State> {
  state: State = {
    weeks: []
  };

  componentWillMount() {
    const { value } = this.props;
    this.setState({ weeks: this.createMonth(value) });
  }

  componentWillReceiveProps(nextProps: Props) {
    if (nextProps.value !== this.props.value) {
      this.setState({ weeks: this.createMonth(nextProps.value) });
    }
  }

  createMonth = (value: Date) => {
    let startDateOfMonth = startOfMonth(value);
    let endDateOfMonth = endOfMonth(value);
    let currentMonth = eachDay(startDateOfMonth, endDateOfMonth);
    let firstOfMonthDay = getDay(startDateOfMonth);
    let beginOfMonth;
    firstOfMonthDay === 0 ? (beginOfMonth = 1) : (beginOfMonth = 8 - firstOfMonthDay);
    let firstWeek = currentMonth.slice(0, beginOfMonth);
    let secondWeek = currentMonth.slice(beginOfMonth, 7 + firstWeek.length);
    let thirdWeek = currentMonth.slice(beginOfMonth + 7, beginOfMonth + 14);
    let fourthWeek = currentMonth.slice(beginOfMonth + 14, beginOfMonth + 21);
    let fiveWeek = currentMonth.slice(beginOfMonth + 21, beginOfMonth + 28);
    let tailWeek = currentMonth.slice(beginOfMonth + 28);
    const weeks = [firstWeek, secondWeek, thirdWeek, fourthWeek, fiveWeek];
    if (tailWeek.length > 0) {
      weeks.push(tailWeek);
    }
    return weeks;
  };

  onSetDate = (date: Date) => {
    const { onSetDate } = this.props;
    if (onSetDate) {
      onSetDate(date);
    }
  };

  onClickMonth = () => {
    const { onClickMonth } = this.props;
    if (onClickMonth) {
      onClickMonth(this.props.value);
    }
  };

  getWeekDays = () => {
    const { weeks } = this.state;
    return weeks[1].map((weekDay, i) => {
      return format(weekDay, 'dd');
    });
  };

  setDayStyle = (highlight: Array<Date> | Date, date: Date) => {
    const highlightStyle = { backgroundColor: '#34495e', color: '#fff' };
    const rangeStyle = { backgroundColor: '#8196ab', color: '#fff' };
    if (Array.isArray(highlight)) {
      for (let i = 0; i < highlight.length; i++) {
        if (isSameDay(highlight[i], date)) {
          return highlightStyle;
        }
      }
      if (highlight[0] < highlight[1]) {
        if (isWithinRange(date, highlight[0], highlight[1])) {
          return rangeStyle;
        }
      } else {
        if (isWithinRange(date, highlight[1], highlight[0])) {
          return rangeStyle;
        }
      }
    } else {
      if (isSameDay(highlight, date)) {
        return highlightStyle;
      }
    }
  };

  render() {
    const { weeks } = this.state;
    const { value, highlight } = this.props;
    return (
      <div className={s.root}>
        <span onClick={this.onClickMonth}>{format(value, 'MMMM YYYY')}</span>
        <table>
          <thead>
            <tr>
              {this.getWeekDays().map((weekDay, i) => {
                return <th key={i}>{weekDay.toUpperCase()}</th>;
              })}
            </tr>
          </thead>
          <tbody>
            {weeks.map((week, i) => (
              <tr key={i}>
                {week.map((date, dayOfWeek) => {
                  return (
                    <CalendarDay
                      key={dayOfWeek}
                      date={date}
                      style={highlight ? this.setDayStyle(highlight, date) : null}
                      onClick={this.onSetDate}
                    />
                  );
                })}
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }
}

export default CalendarMonthGrid;
