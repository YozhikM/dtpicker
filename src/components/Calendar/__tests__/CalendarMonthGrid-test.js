/* @flow */

import React from 'react';
import CalendarMonthGrid from '../CalendarMonthGrid';

describe('CalendarMonthGrid', () => {
  describe('render()', () => {
    it('basic', () => {
      expect(
        shallow(<CalendarMonthGrid value={new Date(2009, 0, 9)} highlight={new Date(2001, 2, 4)} />)
      ).toMatchSnapshot();
    });
  });

  it('componentWillMount()', () => {
    const value = new Date(2019, 0, 9);
    const wrapper = shallow(<CalendarMonthGrid value={value} highlight={new Date()} />);
    wrapper.instance().createMonth(value);
    expect(wrapper.state('weeks')).toMatchSnapshot();
  });

  it('componentWillReceiveProps()', () => {
    const wrapper = shallow(<CalendarMonthGrid value={new Date()} highlight={new Date()} />);
    wrapper.setProps({ value: new Date(2017, 10, 12) });
    expect(wrapper.state('weeks')).toMatchSnapshot();
  });

  it('createMonth()', () => {
    const wrapper = shallow(<CalendarMonthGrid value={new Date()} highlight={new Date()} />);
    const weeks = wrapper.instance().createMonth(new Date(2009, 9, 8));
    expect(weeks).toMatchSnapshot();
  });

  it('getWeekDays()', () => {
    const wrapper = shallow(<CalendarMonthGrid value={new Date()} highlight={new Date()} />);
    const getWeekDays = wrapper.instance().getWeekDays();
    expect(getWeekDays).toMatchSnapshot();
  });

  it('setDayStyle()', () => {
    const wrapper = shallow(
      <CalendarMonthGrid
        value={new Date(1994, 6)}
        highlight={[new Date('1994-07-01T18:00:00.000Z'), new Date('1994-07-05T18:00:00.000Z')]}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('user interactions', () => {
    it('onSetDate()', () => {
      const spy = jest.fn();
      const wrapper = shallow(
        <CalendarMonthGrid value={new Date(2010, 0)} highlight={new Date()} onSetDate={spy} />
      );
      const newDate = new Date(2017, 1, 1);
      wrapper
        .find('CalendarDay')
        .at(0)
        .props()
        .onClick(newDate);
      expect(spy).toBeCalledWith(newDate);
    });

    it('onClickMonth()', () => {
      const spy = jest.fn();
      const value = new Date(2009, 1);
      const wrapper = shallow(
        <CalendarMonthGrid value={value} highlight={new Date()} onClickMonth={spy} />
      );
      wrapper.find('span').simulate('click');
      expect(spy).toBeCalledWith(value);
    });
  });
});
