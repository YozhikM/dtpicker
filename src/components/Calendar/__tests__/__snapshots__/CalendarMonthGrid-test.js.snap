// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarMonthGrid componentWillMount() 1`] = `
Array [
  Array [
    2018-12-31T18:00:00.000Z,
    2019-01-01T18:00:00.000Z,
    2019-01-02T18:00:00.000Z,
    2019-01-03T18:00:00.000Z,
    2019-01-04T18:00:00.000Z,
    2019-01-05T18:00:00.000Z,
  ],
  Array [
    2019-01-06T18:00:00.000Z,
    2019-01-07T18:00:00.000Z,
    2019-01-08T18:00:00.000Z,
    2019-01-09T18:00:00.000Z,
    2019-01-10T18:00:00.000Z,
    2019-01-11T18:00:00.000Z,
    2019-01-12T18:00:00.000Z,
  ],
  Array [
    2019-01-13T18:00:00.000Z,
    2019-01-14T18:00:00.000Z,
    2019-01-15T18:00:00.000Z,
    2019-01-16T18:00:00.000Z,
    2019-01-17T18:00:00.000Z,
    2019-01-18T18:00:00.000Z,
    2019-01-19T18:00:00.000Z,
  ],
  Array [
    2019-01-20T18:00:00.000Z,
    2019-01-21T18:00:00.000Z,
    2019-01-22T18:00:00.000Z,
    2019-01-23T18:00:00.000Z,
    2019-01-24T18:00:00.000Z,
    2019-01-25T18:00:00.000Z,
    2019-01-26T18:00:00.000Z,
  ],
  Array [
    2019-01-27T18:00:00.000Z,
    2019-01-28T18:00:00.000Z,
    2019-01-29T18:00:00.000Z,
    2019-01-30T18:00:00.000Z,
  ],
]
`;

exports[`CalendarMonthGrid componentWillReceiveProps() 1`] = `
Array [
  Array [
    2017-10-31T18:00:00.000Z,
    2017-11-01T18:00:00.000Z,
    2017-11-02T18:00:00.000Z,
    2017-11-03T18:00:00.000Z,
    2017-11-04T18:00:00.000Z,
  ],
  Array [
    2017-11-05T18:00:00.000Z,
    2017-11-06T18:00:00.000Z,
    2017-11-07T18:00:00.000Z,
    2017-11-08T18:00:00.000Z,
    2017-11-09T18:00:00.000Z,
    2017-11-10T18:00:00.000Z,
    2017-11-11T18:00:00.000Z,
  ],
  Array [
    2017-11-12T18:00:00.000Z,
    2017-11-13T18:00:00.000Z,
    2017-11-14T18:00:00.000Z,
    2017-11-15T18:00:00.000Z,
    2017-11-16T18:00:00.000Z,
    2017-11-17T18:00:00.000Z,
    2017-11-18T18:00:00.000Z,
  ],
  Array [
    2017-11-19T18:00:00.000Z,
    2017-11-20T18:00:00.000Z,
    2017-11-21T18:00:00.000Z,
    2017-11-22T18:00:00.000Z,
    2017-11-23T18:00:00.000Z,
    2017-11-24T18:00:00.000Z,
    2017-11-25T18:00:00.000Z,
  ],
  Array [
    2017-11-26T18:00:00.000Z,
    2017-11-27T18:00:00.000Z,
    2017-11-28T18:00:00.000Z,
    2017-11-29T18:00:00.000Z,
  ],
]
`;

exports[`CalendarMonthGrid createMonth() 1`] = `
Array [
  Array [
    2009-09-30T18:00:00.000Z,
    2009-10-01T18:00:00.000Z,
    2009-10-02T18:00:00.000Z,
    2009-10-03T18:00:00.000Z,
  ],
  Array [
    2009-10-04T18:00:00.000Z,
    2009-10-05T18:00:00.000Z,
    2009-10-06T18:00:00.000Z,
    2009-10-07T18:00:00.000Z,
    2009-10-08T18:00:00.000Z,
    2009-10-09T18:00:00.000Z,
    2009-10-10T18:00:00.000Z,
  ],
  Array [
    2009-10-11T18:00:00.000Z,
    2009-10-12T18:00:00.000Z,
    2009-10-13T18:00:00.000Z,
    2009-10-14T18:00:00.000Z,
    2009-10-15T18:00:00.000Z,
    2009-10-16T18:00:00.000Z,
    2009-10-17T18:00:00.000Z,
  ],
  Array [
    2009-10-18T18:00:00.000Z,
    2009-10-19T18:00:00.000Z,
    2009-10-20T18:00:00.000Z,
    2009-10-21T18:00:00.000Z,
    2009-10-22T18:00:00.000Z,
    2009-10-23T18:00:00.000Z,
    2009-10-24T18:00:00.000Z,
  ],
  Array [
    2009-10-25T18:00:00.000Z,
    2009-10-26T18:00:00.000Z,
    2009-10-27T18:00:00.000Z,
    2009-10-28T18:00:00.000Z,
    2009-10-29T18:00:00.000Z,
    2009-10-30T18:00:00.000Z,
  ],
]
`;

exports[`CalendarMonthGrid getWeekDays() 1`] = `
Array [
  "Mo",
  "Tu",
  "We",
  "Th",
  "Fr",
  "Sa",
  "Su",
]
`;

exports[`CalendarMonthGrid render() basic 1`] = `
<div>
  <span
    onClick={[Function]}
  >
    January 2009
  </span>
  <table>
    <thead>
      <tr>
        <th
          key="0"
        >
          MO
        </th>
        <th
          key="1"
        >
          TU
        </th>
        <th
          key="2"
        >
          WE
        </th>
        <th
          key="3"
        >
          TH
        </th>
        <th
          key="4"
        >
          FR
        </th>
        <th
          key="5"
        >
          SA
        </th>
        <th
          key="6"
        >
          SU
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <CalendarDay
          date={2008-12-31T18:00:00.000Z}
          key="0"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-01T18:00:00.000Z}
          key="1"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-02T18:00:00.000Z}
          key="2"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-03T18:00:00.000Z}
          key="3"
          onClick={[Function]}
        />
      </tr>
      <tr
        key="1"
      >
        <CalendarDay
          date={2009-01-04T18:00:00.000Z}
          key="0"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-05T18:00:00.000Z}
          key="1"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-06T18:00:00.000Z}
          key="2"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-07T18:00:00.000Z}
          key="3"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-08T18:00:00.000Z}
          key="4"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-09T18:00:00.000Z}
          key="5"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-10T18:00:00.000Z}
          key="6"
          onClick={[Function]}
        />
      </tr>
      <tr
        key="2"
      >
        <CalendarDay
          date={2009-01-11T18:00:00.000Z}
          key="0"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-12T18:00:00.000Z}
          key="1"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-13T18:00:00.000Z}
          key="2"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-14T18:00:00.000Z}
          key="3"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-15T18:00:00.000Z}
          key="4"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-16T18:00:00.000Z}
          key="5"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-17T18:00:00.000Z}
          key="6"
          onClick={[Function]}
        />
      </tr>
      <tr
        key="3"
      >
        <CalendarDay
          date={2009-01-18T18:00:00.000Z}
          key="0"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-19T18:00:00.000Z}
          key="1"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-20T18:00:00.000Z}
          key="2"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-21T18:00:00.000Z}
          key="3"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-22T18:00:00.000Z}
          key="4"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-23T18:00:00.000Z}
          key="5"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-24T18:00:00.000Z}
          key="6"
          onClick={[Function]}
        />
      </tr>
      <tr
        key="4"
      >
        <CalendarDay
          date={2009-01-25T18:00:00.000Z}
          key="0"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-26T18:00:00.000Z}
          key="1"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-27T18:00:00.000Z}
          key="2"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-28T18:00:00.000Z}
          key="3"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-29T18:00:00.000Z}
          key="4"
          onClick={[Function]}
        />
        <CalendarDay
          date={2009-01-30T18:00:00.000Z}
          key="5"
          onClick={[Function]}
        />
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`CalendarMonthGrid setDayStyle() 1`] = `
<div>
  <span
    onClick={[Function]}
  >
    July 1994
  </span>
  <table>
    <thead>
      <tr>
        <th
          key="0"
        >
          MO
        </th>
        <th
          key="1"
        >
          TU
        </th>
        <th
          key="2"
        >
          WE
        </th>
        <th
          key="3"
        >
          TH
        </th>
        <th
          key="4"
        >
          FR
        </th>
        <th
          key="5"
        >
          SA
        </th>
        <th
          key="6"
        >
          SU
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        key="0"
      >
        <CalendarDay
          date={1994-06-30T17:00:00.000Z}
          key="0"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-01T17:00:00.000Z}
          key="1"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#34495e",
              "color": "#fff",
            }
          }
        />
        <CalendarDay
          date={1994-07-02T17:00:00.000Z}
          key="2"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#8196ab",
              "color": "#fff",
            }
          }
        />
      </tr>
      <tr
        key="1"
      >
        <CalendarDay
          date={1994-07-03T17:00:00.000Z}
          key="0"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#8196ab",
              "color": "#fff",
            }
          }
        />
        <CalendarDay
          date={1994-07-04T17:00:00.000Z}
          key="1"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#8196ab",
              "color": "#fff",
            }
          }
        />
        <CalendarDay
          date={1994-07-05T17:00:00.000Z}
          key="2"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#34495e",
              "color": "#fff",
            }
          }
        />
        <CalendarDay
          date={1994-07-06T17:00:00.000Z}
          key="3"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-07T17:00:00.000Z}
          key="4"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-08T17:00:00.000Z}
          key="5"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-09T17:00:00.000Z}
          key="6"
          onClick={[Function]}
        />
      </tr>
      <tr
        key="2"
      >
        <CalendarDay
          date={1994-07-10T17:00:00.000Z}
          key="0"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-11T17:00:00.000Z}
          key="1"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-12T17:00:00.000Z}
          key="2"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-13T17:00:00.000Z}
          key="3"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-14T17:00:00.000Z}
          key="4"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-15T17:00:00.000Z}
          key="5"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-16T17:00:00.000Z}
          key="6"
          onClick={[Function]}
        />
      </tr>
      <tr
        key="3"
      >
        <CalendarDay
          date={1994-07-17T17:00:00.000Z}
          key="0"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-18T17:00:00.000Z}
          key="1"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-19T17:00:00.000Z}
          key="2"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-20T17:00:00.000Z}
          key="3"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-21T17:00:00.000Z}
          key="4"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-22T17:00:00.000Z}
          key="5"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-23T17:00:00.000Z}
          key="6"
          onClick={[Function]}
        />
      </tr>
      <tr
        key="4"
      >
        <CalendarDay
          date={1994-07-24T17:00:00.000Z}
          key="0"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-25T17:00:00.000Z}
          key="1"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-26T17:00:00.000Z}
          key="2"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-27T17:00:00.000Z}
          key="3"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-28T17:00:00.000Z}
          key="4"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-29T17:00:00.000Z}
          key="5"
          onClick={[Function]}
        />
        <CalendarDay
          date={1994-07-30T17:00:00.000Z}
          key="6"
          onClick={[Function]}
        />
      </tr>
    </tbody>
  </table>
</div>
`;
